### PR TITLE
feat(HACBS-1179) Define Permission Claims for the Integration service…

### PR DESCRIPTION
**Pull request by hongweiliu17**
… in APIExport and APIBinding

Signed-off-by: Hongwei Liu hongliu@redhat.com

### CODE REVIEW

File: config/kcp/apibinding_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      state: Accepted
+    - group: ""
+      resource: "namespaces"
+      state: Accepted
+    - group: ""
+      resource: "configmaps"
+      state: Accepted
+    - resource: "applications"
+      group: "appstudio.redhat.com"
+      identityHash: null # has apiexport
```
Comment (johnbieren) [inline config/kcp/apibinding_integration.yaml:22]: the has and gitops export have combined. See https://github.com/redhat-appstudio/application-api/blob/main/config/kcp/apiexport.yaml

Comment (hongweiliu17) [reply config/kcp/apibinding_integration.yaml:22]: @johnbieren Thank you! Let me update my code.

**Comment by hongweiliu17** [comment]
I meet an error after create CR(component, application, applicationsnapshot)
```
1.664161759907087e+09	ERROR	controllers.snapshot	Failed to get snapshot for	{"Integration": "default/snapshot-sample", "clusterName": "root:users:xr:vd:rh-sso-hongweiliu:hongliu-710", "req": {"namespace": "default", "name": "snapshot-sample"}, "error": "ApplicationSnapshot.appstudio.redhat.com \"snapshot-sample\" not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:320
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234
```

**Comment by johnbieren** [comment]
> I meet an error after create CR(component, application, applicationsnapshot)
> 
> ```
> 1.664161759907087e+09	ERROR	controllers.snapshot	Failed to get snapshot for	{"Integration": "default/snapshot-sample", "clusterName": "root:users:xr:vd:rh-sso-hongweiliu:hongliu-710", "req": {"namespace": "default", "name": "snapshot-sample"}, "error": "ApplicationSnapshot.appstudio.redhat.com \"snapshot-sample\" not found"}
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
> 	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
> 	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:320
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
> 	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
> 	/Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234
> ```
> 
> Then I add some log as below to my local /Users/hongliu/HAS/integration-permissionclaim/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121 and try to log something
> 
> ```
> 	log := logf.FromContext(ctx)
> 	log.Info(fmt.Sprintf("ctx, req", ctx, req))
> 	return c.Do.Reconcile(ctx, req)
> ```
> 
> The info is logged out as
> 
> ```
> 1.664162732657978e+09	INFO	ctx, req%!(EXTRA *context.valueCtx=context.Background.WithCancel.WithValue(type logr.contextKey, val <not Stringer>), reconcile.Request=default/snapshot-sample)	{"controller": "applicationsnapshot", "controllerGroup": "appstudio.redhat.com", "controllerKind": "ApplicationSnapshot", "applicationSnapshot": {"name":"snapshot-sample","namespace":"default"}, "namespace": "default", "name": "snapshot-sample", "reconcileID": "c99c870e-f645-4dde-ab91-28a48e4978dd"}
> ```

what introduced this error? It is hard to narrow it down since this PR is so large

**Comment by hongweiliu17** [comment]
> 

This issue is fixed. I should use `github.com/kcp-dev/logicalcluster/v2` not v1. Thank you!

**Review by dirgim — APPROVED** [review]
LGTM

**Review by johnbieren — APPROVED** [review]
lgtm. Note that you are going to have to add another copy of the apiexport to infra-deployments. I will tag you on the PR when I do it.. I am working on it now. It is due to the identity hashes (I think it is not ideal because it is code duplicated in multiple spots, but)

**Comment by hongweiliu17** [comment]
> lgtm. Note that you are going to have to add another copy of the apiexport to infra-deployments. I will tag you on the PR when I do it.. I am working on it now. It is due to the identity hashes (I think it is not ideal because it is code duplicated in multiple spots, but)

Thank you very much!

**Pull request merged**